src/act_info.cpp

```diff
@@ -1620,22 +1624,19 @@ void do_whois(CHAR_DATA *ch, const char *argument) {
 
     output[0] = '\0';
 
-    for (d = descriptor_list; d != nullptr; d = d->next) {
-        char const *class_name;
-
-        if (!d->is_playing() || !can_see(ch, d->character()))
-            continue;
+    for (auto &d : descriptors().all_visible_to(ch)) {
 
-        auto *wch = d->person();
+        auto *wch = d.person();
 
+        // TODO: can or should this be part of all_visible_to?
```

Comment:
I think we'll have to be wary of having too much of those canned filters, lest it become hard to tell which one to pick :) I guess it's all in the name.

Reply:
I guess if we had a good selection of single-step filters, we can chain them ourselves too.

Reply:
Right. Thoughts and ideas welcomed here. I don't have a strong opinion: where I saw the same kind of check used in more than one place I extracted a canned filter. With a good name we're probably ok. With chained filters one has to remember the right sequence oneself (like `for (auto &x : descs.all() | playing | not_same(ch) | visible_to(ch))` vs `for (auto &x : descs.visible_to(ch))`

Composition is super cool but I'd imagine we'd find ourselves repeating the same pipeline list multiple times and then we'd make a helper function to do the work for us.

Very much open to trying this as well though!

Reply:
I suspect simple method names is probably cleanest (and more reusable).  At the extreme, you could end up with NxM different combinations though hehe.